lib/schemapack.js

```diff
@@ -273,7 +276,7 @@ class SchemaPack {
         return callback(error);
       }
 
-      this.openapi = newOpenapi;
+      this.openapi = deref.dereferenceByConstraint(newOpenapi, getNegativeRegexp('#/components/'));
```

Comment:
Here we are dereferencing only the elements that are referenced from out of '#/components'.

Reply:
Same here. i.e. add comment in code explaining situation.